### DISCUSSION TITLE
Best way to capture a memory dump of executing benchmarks

### DISCUSSION THREAD

**Discussion by paulomorgado** [dotnet/BenchmarkDotNet#2636] [Q&A]
I'm running into errors when running benchmarks, while all other tests run fine.

There's some fuzziness in the test data and it's hard to reproduce.

What would be the best way to capture a memory dump when an exception occurs?

ProcDump will capture dumps of the main process.

If I use `Debugger.Launch(); Debugger.Break();`, I'll be prompted for a debugger. But that's too manual.

**Comment by leonvandermeer** [comment]
[Have a look at my code](https://github.com/leonvandermeer/on-my-route/blob/8c8224a61fc6b3c5c0c0a0dd8a0afe3664cc39fa/OnMyRoute/CrashHandler.cs) where I create a dump when unhandled exception occurs.
Copy and adjust for your needs.
The actual dump creation is done using `Microsoft.Diagnostics.NETCore.Client` nuget package
